apps/mobile/src/components/browser/browser-sheet/url-input.tsx

```diff
@@ -0,0 +1,285 @@
+import { ReactNode, useMemo, useRef } from 'react';
```

Comment:
Looks like this component could use a little refactoring–mostly naming & organization issues.

Few glaring ones:
* There's `searchBarStyle` and `SearchBarButton` but nothing explicitly named search bar? Perhaps `SearchBarButton -> ToolbarButton`, and extract the Toolbar into its own component?
* `UrlInputContainer`doesn't seem to contain the input
* Unlike SearchBarButton, Pressables inside UrlInputContainer aren't named 

A round or two of "moving things around until it feels right" should make this much easier to reason about 👌